node/src/reactor/main_reactor/keep_up.rs

```diff
@@ -597,7 +597,7 @@ impl MainReactor {
                 if block_header.is_genesis() {
                     return Ok(Some(SyncBackInstruction::GenesisSynced));
                 }
-                if self.sync_back_is_ttl() {
+                if self.sync_back_is_ttl()? {
```

Comment:
nit: I think it would have been more readable to have this function named `synced_to_ttl()` or something similar and do the test for the config here. This way the function would have a single purpose and we could get rid of the config test inside the function. But this is a personal preference and not blocking.
```suggestion
                if !self.sync_to_genesis && self.synced_to_ttll()? {
```

Reply:
That's a good suggestion. Update here: b404e158c59208f64f6b5bec11df01758323f5bb